pkglistgen.py

```diff
@@ -0,0 +1,622 @@
+        self.recommends = dict()
+        for p in recommends:
+            already_present = False
+            for m in modules:
+                for arch in ('*', ) + ARCHITECTURES:
+                    already_present = already_present or (p in m.solved_packages[arch])
+                    already_present = already_present or (p in m.recommends)
+            if not already_present:
+                self.recommends[p] = recommends[p]
+
+    def toxml(self, arch):
+        packages = self.solved_packages[arch]
+
+        name = self.name
+        if arch != '*':
+            name += '.' + arch
+
+        root = ET.Element('group', {'name': name})
+        c = ET.Comment(' ### AUTOMATICALLY GENERATED, DO NOT EDIT ### ')
+        root.append(c)
+
+        if arch != '*':
+            cond = ET.SubElement(root, 'conditional', {
+                                 'name': 'only_{}'.format(arch)})
+        packagelist = ET.SubElement(
+            root, 'packagelist', {'relationship': 'recommends'})
+
+        missing = dict()
+        if arch == '*':
+            missing = self.not_found
+        unresolvable = self.unresolvable.get(arch, dict())
+        for name in sorted(packages.keys() + missing.keys() + unresolvable.keys()):
+            if name in self.silents:
+                continue
+            if name in missing:
+                c = ET.Comment(' {} not found on {}'.format(name, ','.join(sorted(missing[name]))))
+                packagelist.append(c)
+                continue
+            if name in unresolvable:
+                c = ET.Comment(' {} uninstallable: {}'.format(name, unresolvable[name]))
+                packagelist.append(c)
+                continue
+            status = self.pkglist.supportstatus(name)
+            p = ET.SubElement(packagelist, 'package', {
+                'name': name,
+                'supportstatus': status})
+            c = ET.Comment(' reason: {} '.format(packages[name]))
+            packagelist.append(c)
+        if arch == '*' and self.develpkgs:
+            c = ET.Comment("\nDevelopment packages:\n  - " + "\n  - ".join(sorted(self.develpkgs)) + "\n")
+            root.append(c)
+        if arch == '*' and self.recommends:
+            comment = "\nRecommended packages:\n"
+            for p in sorted(self.recommends.keys()):
+                comment += "  - {} # {}\n".format(p, self.recommends[p])
+            c = ET.Comment(comment)
+            root.append(c)
+
+        return root
+
+    def dump(self):
+        pprint({'name': self.name, 'missing': self.missing, 'packages': self.packages,
+                'solved': self.solved_packages, 'silents': self.silents})
+        return
+        archs = ('*',) + ARCHITECTURES
+        for arch in archs:
+            x = self.toxml(arch)
+            print(ET.tostring(x, pretty_print=True))
+
+
+class PkgListGen(ToolBase.ToolBase):
+
+    def __init__(self, repostr):
+        ToolBase.ToolBase.__init__(self)
```

Comment:
ToolBase.ToolBase is new style class so `super(PkgListGen, self).__init__()` is better